### ISSUE TITLE
Downloading dependencies under proxy

### ISSUE THREAD

**Issue by aghasemi** [jbangdev/jbang#1010]
Hi,

JBang cannot download dependencies when under corporate proxy. Interestingly, JDK is downloaded without a problem. I guess it is because JDK and dependencies are downloaded through different channels (JBang proper vs Maven?) where one reads http_* environment variables and the other doesn't. Is there an environment variable I can set to fix this for apparently Maven, that is used under the hood for downloading dependencies? MAVEN_OPTS and JAVA_TOOL_OPTS did not work, and I prefer not to create a `settings.xml` file in the Maven home.

Many thanks

**Comment by maxandersen** [comment]
does it work when you set it via settings.xml? if that does not work then no amount of env var to change it will fix it.

users reports JAVA_TOOL_OPTS works here https://github.com/jbangdev/jbang/discussions/671
or system proxies https://github.com/jbangdev/jbang/issues/547#issuecomment-734154630

**Comment by patric-r** [comment]
jbang's proxy support looks broken.

`As org.apache.maven.wagon.providers.http.LightweightHttpsWagon.openConnection()` is actually used for making the HTTP connection (via sun.net HttpClient), it ignores system proxy configuration.
Instead, it looks for the proxy configuration within the `org.eclipse.aether.repository.RemoteRepository.RemoteRepository`
but jbang fails to set a proxy in the Repository's builder.

**Comment by quintesse** [comment]
I can confirm that the two solutions mentioned before do not work.

Thanks @patric-r for the feedback, that's very useful!
You're not able to provide a fix by any chance as well, are you? :wink:

**Comment by quintesse** [comment]
Btw, JBang doesn't deal directly with the Aether API but uses [Shrinkwrap](https://github.com/shrinkwrap/resolver), so we have to figure out how to set a proxy for that.

**Comment by patric-r** [comment]
@quintesse Yes, I noticed that as well. IMHO dev.jbang.dependencies.DependencyUtil.resolveDependenciesViaAether() looks like a good start for dealing with the proxy.

For sensible defaults, it make sense to either use the system proxy or maven defaults (e.g. .m2/settings.xml).
BTW, is there a reason why jbang uses a customized version of shrinkwrap? This feels a little bit "unusual" (not to say hacky...).

**Comment by quintesse** [comment]
> looks like a good start for dealing with the proxy.

Indeed, the problem is _how_. There doesn't seem to be anything in the API that deals with Proxies (at least the last time I looked).

> is there a reason why jbang uses a customized version of shrinkwrap? 

The explanation for that can be found in this PR: https://github.com/jbangdev/jbang/pull/673

**Comment by quintesse** [comment]
I also just doubly verified that setting the proxy in `settings.xml` makes the `mvn` command use it but it does not affect the behavior of `jbang`.

**Comment by gwentey** [comment]
same issue here

**Comment by maxandersen** [comment]
Which issue? Settings.xml not working or?

**Comment by quintesse** [comment]
Any issues with proxies should go away once we decide to switch to MIMA.

**Comment by aghasemi** [comment]
> Any issues with proxies should go away once we decide to switch to MIMA.

What is MIMA? :)

**Comment by maxandersen** [comment]
MIMA is what is coming in via this PR https://github.com/jbangdev/jbang/pull/1604

It is where jbang will be able to use more recent setup of maven's artifact resolver.

**Comment by quintesse** [comment]
Yeah, but later I realized that will only fix the proxy issues for artifact resolving/downloading, not for the other http requests we make. So that's still something we should definitely fix in our code.

**Comment by maxandersen** [comment]
@quintesse remind me - did we or did we not fix this yet ? ...and didn't we have nice way of testing it?

**Comment by quintesse** [comment]
I'm not sure anymore, it's been a while.
We do have a way to test, but I wouldn't say it's necessarily "nice" 😅 
You can use this to run JBang in a container that doesn't allow direct network access https://github.com/jbangdev/jbang/tree/main/misc/proxy
